app/src/components/ProposalList.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import List from '@material-ui/core/List';
+import ProposalItem from './ProposalItem';
+import { withStyles } from '@material-ui/core/styles';
+import Styles from '../theme';
+
+interface ProposalListProps {
+  drizzle: any;
+  drizzleState: any;
+  classes: any;
+}
+
+type ProposalListState = {
+  ids: any;
+}
+
+class ProposalList extends React.Component<ProposalListProps, ProposalListState> {
+  componentDidMount () {
+    this.props.drizzle.contracts.Governance.methods.getProposals().call()
```

Comment:
probably ought to split this into a variable and (e.g. governanceContract), and then calling the stuff on that variable

i also think when you do method chaining like this you should linebreak on each method, but idk if the linter allows it

Reply:
I agree about method chaing, as for splitting out into a seperate variable, I do not know enough about typescript or react to be able to work out how to do that, I've attempted it but I keep getting state errors so am going to ignore the contract variable comments